Source/Task/iOS/ios_WaitTimer.mm

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pch.h"
+
+#include "WaitTimer.h"
+
+#import <Foundation/Foundation.h>
+
+#include "ios_WaitTimerImpl.h"
+
+using Deadline = std::chrono::high_resolution_clock::time_point;
+
+WaitTimerImpl::WaitTimerImpl()
+: m_context(nullptr),
+  m_callback(nullptr),
+  m_target(nullptr),
+  m_timer(nullptr)
+{
+}
+
+WaitTimerImpl::~WaitTimerImpl()
+{
+    if (m_timer != nullptr && m_timer.valid)
+    {
+        [m_timer invalidate];
+    }
+}
+
+HRESULT WaitTimerImpl::Initialize(_In_opt_ void* context, _In_ WaitTimerCallback* callback){
+    m_context = context;
+    m_callback = callback;
+    m_target = [ios_WaitTimer_target new];
+    return S_OK;
+}
+
+void WaitTimerImpl::Start(_In_ uint64_t absoluteTime)
+{
+    Cancel();
+    
+    auto duration = Deadline::duration(absoluteTime);
+    auto timePoint = Deadline(duration) - std::chrono::high_resolution_clock::now();
+
+    m_timer = [NSTimer scheduledTimerWithTimeInterval:timePoint.count() / 1000.0
```

Comment:
can you use a std::duration_cast to convert to the correct units without making assumptions about the representation of Deadline?

Reply:
Will do. Separate push since I will want to verify on my mac.